### PR TITLE
vSphere docs: add IPI documentation

### PR DESCRIPTION
**Pull request by patrickdillon**
Brings vSphere IPI docs up to par with other platforms.

/cc @jcpowermac 
/cc @dav1x 
/cc @abhinavdahiya 

@dav1x please feel free to invite anyone from vmware

cc @gmarkley-VI @aravindhp @codyhoag

### CODE REVIEW

File: docs/user/vsphere/requirements.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Resource Requirements
```
Comment (jcpowermac) [inline docs/user/vsphere/requirements.md:1]: vSphere IPI requires DHCP

File: docs/user/vsphere/post.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# (Optional) Unassign Privileges
+Once an install has completed, the cluster does not require any user privileges. If a user was granted adminstrative or escalated privileges, those privileges can be safely removed, if desired. Keep in mind, that the privileges would be required in order to destroy the cluster.
```
Comment (jcpowermac) [inline docs/user/vsphere/post.md:2]: The user that you are providing for installation will also be used for cloud-config.  That user needs to create disks for PVs. You won't be able to remove it.

https://vmware.github.io/vsphere-storage-for-kubernetes/documentation/vcp-roles.html

File: docs/user/vsphere/requirements.md (hunk 1 above)
Comment (dav1x) [reply docs/user/vsphere/requirements.md:1]: I haven't reviewed the keepalived config recently. But, we added an issue multicast VRRP creating issues with multiple clusters on the same broadcast domain. https://issues.redhat.com/browse/KNIDEPLOY-1978

File: docs/user/vsphere/privileges.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+Here is an example summary of privileges that could be used to install a cluster:
+
+- __Datastore__
+  - _Allocate space_
+- __Folder__
+  - _Create folder_
+  - _Delete folder_
+- __Host__
+  - All privileges
```
Comment (jcpowermac) [inline docs/user/vsphere/privileges.md:19]: During 4.6 we should get more clarification (I can help) with what permissions are required.  Host - all privileges seems very broad.

File: docs/user/vsphere/install.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+IPI install.
+
+To see a guided example of a UPI install, see [Install: User-Provided Infrastructure](install_upi.md)
+
+## Example: Installer-Provided Infrastructure (IPI)
+
+The steps for performing an IPI-based install are outlined [here][cloud-install]. Following this guide you may begin at
+the Download step.
+
+### Trust Root CA
```
Comment (abhinavdahiya) [inline docs/user/vsphere/install.md:14]:
```suggestion
### Trust vCenter API
```

File: docs/user/vsphere/install.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+The `create install-config`, `create cluster`, and `destroy cluster` commands access the vCenter API. The vCenter's root CA certificates must be added to the system trust before connecting to the API. The certs can be downloaded from the vCenter's homepage:
```
Comment (abhinavdahiya) [inline docs/user/vsphere/install.md:15]: >  The `create install-config`, `create cluster`, and `destroy cluster` commands access the vCenter API.

I think enumerating these commands is over stitching, there is a change any command you run will access the vCenter API to validate and prevent failures.

File: docs/user/vsphere/install.md [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+Add the certs appropriate for your OS to your system trust. For Fedora:
+
+```console
+sudo cp certs/lin/* /etc/pki/ca-trust/source/anchors
+sudo update-ca-trust extract
+```
+### Create Configuration
+
+```console
+? SSH Public Key /home/user_id/.ssh/id_rsa.pub
```
Comment (abhinavdahiya) [inline docs/user/vsphere/install.md:55]: this missing the openshift-install invocation.

File: docs/user/vsphere/install.md [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+INFO Waiting up to 30m0s for bootstrapping to complete...
+INFO Destroying the bootstrap resources...
+INFO Waiting up to 30m0s for the cluster at https://api.mycluster.example.com:6443 to initialize...
+INFO Waiting up to 10m0s for the openshift-console route to be created...
+INFO Install complete!
+INFO To access the cluster as the system:admin user when using 'oc', run 'export KUBECONFIG=/home/user/auth/kubeconfig'
+INFO Access the OpenShift web-console here: https://console-openshift-console.apps.mycluster.example.com
+INFO Login to the console with user: kubeadmin, password: 5char-5char-5char-5char
+```
+
```
Comment (abhinavdahiya) [inline docs/user/vsphere/install.md:89]: i think we should include https://github.com/openshift/installer/blob/master/docs/user/gcp/install.md#running-cluster for vSphere cluster. this includes a short few paragraphs on what's created by the installer.

**Comment by abhinavdahiya** [comment]
/approve

just a few nits, otherwise this looks good.

**Comment by dav1x** [comment]
cc @RobbieJVMW

File: docs/user/vsphere/requirements.md (hunk 1 above)
Comment (patrickdillon) [reply docs/user/vsphere/requirements.md:1]: I updated. Please take a look if more detail should be provided or if this sounds wrong.

File: docs/user/vsphere/privileges.md (hunk 3 above)
Comment (patrickdillon) [reply docs/user/vsphere/privileges.md]: I was able to remove these permissions entirely and successfully install. This must have slipped through my trial-and-error process. I am running another test now, just to make sure.

File: docs/user/vsphere/privileges.md [hunk 8]
```diff
@@ -0,0 +1,50 @@
+# Required Privileges
+In order to install an OpenShift cluster to a vCenter, the user provided to the installer needs privileges to read and create the necessary resources. The easiest way to achieve this level of permission and ensure success is to install with a user who has administrative privileges.
+
+If the provided user has admin privileges, no action is required and you can skip to the next step. Otherwise, the rest of this document can be used as a resource to create a user with more fine-grained privileges.
+
+## Privileges
+In order to create an OpenShift cluster, a user needs permissions for the following categories: _Datastore_, _Folder_, _Host_, _vSphere Tagging_, _Network_, _Resource_, _Profile-driven storage_, _vApp_, and _Virtual machine_.
```
Comment (dav1x) [inline docs/user/vsphere/privileges.md:7]: That same user will be assigned as the cloud provider account as well. We should probably align the required permission to the dynamic provisioning docs: https://vmware.github.io/vsphere-storage-for-kubernetes/documentation/vcp-roles.html

**Pull request merged**